lib/chef/provider/apt_update.rb

```diff
@@ -75,6 +77,13 @@ def do_update
         declare_resource(:execute, "apt-get -q update")
       end
 
+      def self.uses_apt?
+        ENV["PATH"] ||= ""
+        paths = %w{ /bin /usr/bin /sbin /usr/sbin } + ENV["PATH"].split(::File::PATH_SEPARATOR)
+        paths.any? { |path| ::File.executable?(::File.join(path, "apt-get")) }
```

Comment:
yeah, i'm copying what the apt_repository provider does, i'm not entirely sure why @thommay wasn't using that, but the code is slightly different here.